plugin/pkg/dnsutil/reverse.go

```diff
@@ -33,12 +33,13 @@ func ExtractAddressFromReverse(reverseName string) string {
 // name is in a reverse zone. The returned integer will be 1 for in-addr.arpa. (IPv4)
 // and 2 for ip6.arpa. (IPv6).
 func IsReverse(name string) int {
-	if strings.HasSuffix(name, IP4arpa) {
+	if strings.HasSuffix(name, IP4arpa) || name == IP4arpaDom {
 		return 1
 	}
-	if strings.HasSuffix(name, IP6arpa) {
+	if strings.HasSuffix(name, IP6arpa) || name == IP6arpaDom {
```

Comment:
this should be needed.

Reply:
?

Reply:
Do you mean, "This should _not_ bet needed." ?

This does fix a bug in the existing code, where `in-addr.arpa.` is not identified as a reverse domain.
This can cause the _kubernetes_ plugin to select a reverse domain as its primary domain if `in-addr.arpa` is listed before `cluster.local`. e.g. `kubernetes in-addr.arpa cluster.local`

Reply:
Hmm. OK.  I'll create a different function that does what I need, and remove IsReverse(), since it will no longer be used by anything.

Reply:
... actually - i'll create a new function that only checks the domain, and call both.

Reply:
Ive added a new function, and also call that when i'm comparing zones that may be a base reverse domain.